lapis/src/test/kotlin/org/genspectrum/lapis/model/mutationsOverTime/AminoAcidMutationsOverTimeModelTest.kt

```diff
@@ -0,0 +1,254 @@
+        mockSiloAminoAcidCoverageQuery(siloQueryClient, "S", 1, DUMMY_DATE_BETWEEN_ALL, Stream.empty())
+
+        val mutations = listOf(DUMMY_MUTATION1)
+        val dateRanges = listOf(DUMMY_DATE_RANGE1, DUMMY_DATE_RANGE2)
+
+        val result = underTest.evaluateAminoAcidMutations(
+            mutations = mutations,
+            lapisFilter = DUMMY_LAPIS_FILTER,
+            dateField = DUMMY_DATE_FIELD,
+            dateRanges = dateRanges,
+        )
+
+        assertThat(result.mutations, equalTo(mutations.map { it.toString(referenceGenome) }))
+        assertThat(result.dateRanges, equalTo(dateRanges))
+        assertThat(
+            result.data,
+            equalTo(
+                listOf(
+                    listOf(MutationsOverTimeCell(0, 0), MutationsOverTimeCell(0, 0)),
+                ),
+            ),
+        )
+    }
+
+    @Test
+    fun `given one data version change, then it succeeds`() {
+        var callCount = 0
+        every {
+            siloQueryClient.sendQueryAndGetDataVersion<AggregationData>(any(), false)
+        } answers {
+            val version = if (callCount++ == 0) "1" else "2"
+            WithDataVersion(
+                version,
+                Stream.of(
+                    AggregationData(1, fields = mapOf(DUMMY_DATE_FIELD to TextNode("2021-06-01"))),
+                ),
+            )
+        }
+
+        val mutations = listOf(DUMMY_MUTATION1)
+        val dateRanges = listOf(DUMMY_DATE_RANGE1)
+
+        val result = underTest.evaluateAminoAcidMutations(
+            mutations = mutations,
+            lapisFilter = DUMMY_LAPIS_FILTER,
+            dateField = DUMMY_DATE_FIELD,
+            dateRanges = dateRanges,
+        )
+
+        assertThat(result.data[0][0].count, equalTo(1))
+    }
+
+    @Test
+    fun `given more than once data version change, then it throws`() {
+        var callCount = 0
+        every {
+            siloQueryClient.sendQueryAndGetDataVersion<AggregationData>(any(), false)
+        } answers {
+            val version = (callCount++).toString()
+            WithDataVersion(
+                version,
+                Stream.of(
+                    AggregationData(1, fields = mapOf(DUMMY_DATE_FIELD to TextNode("2021-06-01"))),
+                ),
+            )
+        }
+
+        val mutations = listOf(DUMMY_MUTATION1)
+        val dateRanges = listOf(DUMMY_DATE_RANGE1)
+
+        val exception = assertThrows<RuntimeException> {
+            underTest.evaluateAminoAcidMutations(
+                mutations = mutations,
+                lapisFilter = DUMMY_LAPIS_FILTER,
+                dateField = DUMMY_DATE_FIELD,
+                dateRanges = dateRanges,
+            )
+        }
+        assertThat(exception.message, containsString("data has been updated multiple times"))
+    }
+
+    @Test
+    fun `given a maybe() query, then it throws`() {
+        val mutations = listOf(AminoAcidMutation("S", 1, "R", true))
```

Comment:
Nitpick: Especially on boolean arguments using named arguments improves readability:
```suggestion
        val mutations = listOf(AminoAcidMutation("S", 1, "R", maybe = true))
```